yf_downloader.py

```diff
@@ -0,0 +1,170 @@
+
+    def download_minimum_interval(self, ticker : str, start_date : date, end_date : date):
+        '''
+        Downloads quote data for a single ticker given the start date and end date, using the smallest interval
+        available for the given dates.
+
+        Parameters:
+            ticker : str
+                The simbol to download data of.
+            start_datetime : datetime
+                Must be before end_datetime.
+            end_datetime : datetime
+                Must be after start_datetime.
+        '''
+        return self.download_dates(ticker, start_date, end_date, __calc_minimum_interval(start_date))
+
+
+    def download_last_week(self, ticker : str):
+        '''Downloads quote data for a single ticker a puts it in a beautified JSON file.
+
+        Parameters:
+            ticker : str
+                The symbol to download data of. Could be with market indication (FTSEMIB) or with market (FTSEMIB.MI).
+        '''
+
+        today = date.today()
+        seven_days_ago = today - timedelta(days=7)
+
+        return self.download_minimum_interval(ticker, seven_days_ago, today)
+
+    def __write_on_JSON_file(self, json_filename : str, yf_data : DataFrame):
+        '''
+        Creates a new file in the files_directory and fills it with tabled Dataframe
+
+        Parameters:
+            json_filename : str
+            yf_data : DataFrame
+                Downloaded data with yf.download()
+        '''
+        json_filepath = Path(self.files_directory, json_filename)
+        json_filepath.open("w+").write(yf_data.to_json(orient="table", indent=4))
+        return json_filepath
+
+# Functions
+
+def __yahoo_time_format(date : date):
+    '''
+    Formats time into yfinance-ready string format for start date and end date.
+
+    Parameters:
+        date : datetime.date
+            Date to be formatted for yfinance start and end times.
+    '''
+    return date.strftime("%Y-%m-%d")
+
+def __output_JSON_filename_dates(ticker : str, start_date : date, end_date : date):
+    '''
+    Generates the JSON output file's name using ticker, data start date and end date.
+
+    Parameters:
+        ticker : str
+            Output ticker's name.
+        start_datetime : datetime
+            Beginning date of output data.
+        end_datetime : datetime
+            End date of output data.
+    '''
+    return "{}_{}_to_{}.json".format(ticker, end_date.strftime("%d-%m-%y"), start_date.strftime("%d-%m-%y"))
+
+def __output_JSON_filename_period(ticker : str, period : str):
+    '''
+    Generates the JSON output file's name using ticker and period of time.
+
+    Parameters:
+        ticker : str
+            Output ticker's name.
+        period : str
+            Yahoo's period of time
+    '''
+    return "{}_{}_{}.json".format(ticker, date.today().strftime("%d-%m-%y"), period)
+
+def __calc_minimum_interval(start_date : date):
```

Comment:
I'm gonna make a wild guess and assume the double underscores mean "private". TIL

Reply:
These are private functions, they used to be private methods but made no use of the "self" parameter, so codacy suggested to move them to functions. I have no idea if there are such things as private and public functions of a module but I'll find out.